src/sinks/util/http.rs

```diff
@@ -21,11 +21,64 @@ pub struct HttpService {
 }
 
 impl HttpService {
+    pub fn builder() -> HttpServiceBuilder {
+        HttpServiceBuilder::new()
+    }
+
     pub fn new<F>(request_builder: F) -> Self
     where
         F: Fn(Vec<u8>) -> hyper::Request<Vec<u8>> + Sync + Send + 'static,
     {
-        let https = HttpsConnector::new(4).expect("TLS initialization failed");
+        Self::builder().build(request_builder)
+    }
+}
+
+/// A builder for `HttpService`s
+pub struct HttpServiceBuilder {
```

Comment:
Generally, I name these just `Builder` and they can only be constructed from some `builder` fn on the main struct.

Reply:
The `new` method is private, so it can only be built by code in this module.

Reply:
ah good catch, I missed that 😄